File: 351002/Stepanenko/src/main/java/by/bsuir/distcomp/mapper/MarkerMapper.java
package by.bsuir.distcomp.mapper;

import by.bsuir.distcomp.dto.request.MarkerRequestTo;
import by.bsuir.distcomp.dto.response.MarkerResponseTo;
import by.bsuir.distcomp.entity.Marker;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface MarkerMapper {
    Marker toEntity(MarkerRequestTo dto);

    MarkerResponseTo toResponseDto(Marker entity);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/MarkerMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.marker.MarkerRequestTo
import com.distcomp.dto.marker.MarkerResponseTo
import com.distcomp.entity.Marker
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface MarkerMapper {
    fun toMarkerResponse(marker: Marker) : MarkerResponseTo

    fun toMarkerEntity(markerRequestTo: MarkerRequestTo) : Marker
}